cli/src/alluxio.org/cli/processes/master.go

```diff
@@ -0,0 +1,170 @@
+
+	masterJavaOpts += envVar.GetString(env.ConfAlluxioJavaOpts.EnvVar)
+	masterJavaOpts += envVar.GetString(p.JavaOptsEnvVarKey)
+
+	envVar.Set(p.JavaOptsEnvVarKey, strings.TrimSpace(masterJavaOpts)) // leading spaces need to be trimmed as a exec.Command argument
+}
+
+func (p *MasterProcess) StartCmd(cmd *cobra.Command) *cobra.Command {
+	cmd.Use = p.Name
+	cmd.Flags().StringVar(&p.JournalBackupFile, "journal-backup-file", "", "Path to journal backup file to restore the master from")
+	return cmd
+}
+
+func (p *MasterProcess) Start(cmd *env.StartProcessCommand) error {
+	if err := p.checkJournal(); err != nil {
+		return stacktrace.Propagate(err, "error validating journal")
+	}
+
+	cmdArgs := []string{env.Env.EnvVar.GetString(env.ConfJava.EnvVar)}
+	if attachOpts := env.Env.EnvVar.GetString(confAlluxioMasterAttachOpts.EnvVar); attachOpts != "" {
+		cmdArgs = append(cmdArgs, strings.Split(attachOpts, " ")...)
+	}
+	cmdArgs = append(cmdArgs, "-cp", env.Env.EnvVar.GetString(env.EnvAlluxioServerClasspath))
+
+	masterJavaOpts := env.Env.EnvVar.GetString(p.JavaOptsEnvVarKey)
+	cmdArgs = append(cmdArgs, strings.Split(masterJavaOpts, " ")...)
+
+	if p.JournalBackupFile != "" {
+		cmdArgs = append(cmdArgs, strings.TrimSpace(fmt.Sprintf(env.JavaOptFormat, confAlluxioMasterJournalInitFromBackup, p.JournalBackupFile)))
+	}
+	// specify a default of -Xmx8g if no memory setting is specified
+	const xmxOpt = "-Xmx"
+	if !strings.Contains(masterJavaOpts, xmxOpt) && !strings.Contains(masterJavaOpts, "MaxRAMPercentage") {
+		cmdArgs = append(cmdArgs, fmt.Sprintf("%v8g", xmxOpt))
+	}
+	// specify a default of -XX:MetaspaceSize=256M if not set
+	const metaspaceSizeOpt = "-XX:MetaspaceSize"
+	if !strings.Contains(masterJavaOpts, metaspaceSizeOpt) {
+		cmdArgs = append(cmdArgs, fmt.Sprintf("%v=256M", metaspaceSizeOpt))
+	}
+
+	cmdArgs = append(cmdArgs, p.JavaClassName)
+
+	if err := p.Launch(cmd, cmdArgs); err != nil {
+		return stacktrace.Propagate(err, "error launching process")
+	}
+	return nil
+}
+
+func (p *MasterProcess) StopCmd(cmd *cobra.Command) *cobra.Command {
+	cmd.Use = p.Name
+	return cmd
+}
+
+func (p *MasterProcess) checkJournal() error {
+	journalDir, err := conf.Get.FetchValue(confAlluxioMasterJournalFolder)
+	if err != nil {
+		return stacktrace.Propagate(err, "error fetching value for %v", confAlluxioMasterJournalFolder)
+	}
+	stat, err := os.Stat(journalDir)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return stacktrace.Propagate(err, "error listing path at %v", journalDir)
+	}
+	if !stat.IsDir() {
+		return stacktrace.NewError("Journal location %v is not a directory. Please remove the file and retry.", journalDir)
+	}
+	isEmpty, err := dirIsEmpty(journalDir)
+	if err != nil {
+		return stacktrace.Propagate(err, "error listing contents of %v", journalDir)
+	}
+	if !isEmpty {
+		// TODO: do we always format the journal on master startup for embedded journal? this is strange but following the previous logic
```

Comment:
@jenoudet question for you; could you take a look at the existing logic and confirm this?

Reply:
see https://github.com/Alluxio/alluxio/blob/3f9fd645fa7a30a5a68b08843c108d1653d92e80/bin/alluxio-start.sh#LL209C4-L209C4

if format flag is not specified and journal type is embedded journal and journal folder is a directory and journal folder is not empty, run format master

Reply:
https://github.com/Alluxio/alluxio/blob/3c576b7bdd669d688dc2a343aa74813b7bf3f98e/dora/core/server/common/src/main/java/alluxio/cli/Format.java#L113-L122 
`bin/alluxio format/formatMasters` end up running `bin/alluxio formatJournal`, which just empties the journal folder, so formatting it if empty is useless and cyclical and can therefore be omitted.